datalad/support/network.py

```diff
@@ -235,3 +249,503 @@ def __call__(self, url):
             return dict(mtime=r_stamp['mtime'], size=r_stamp['size'], url=url)
         finally:
             r.close()
+
+
+# TODO:  make it consistent/clear at what stage % encoding/decoding happens!
+# now it is a mix!
+
+#
+# Useful functionality in requests.models
+#  utils.requote_uri -- quote/unquote cycle to guarantee consistent appearance
+#  RequestEncodingMixin._encode_params -- Will successfully encode parameters when passed as a dict or a list of ...
+#  PreparedRequest().prepare_url(url, params) -- nicely cares about url encodings etc
+#
+
+def _guess_ri_cls(ri):
+    """Factory function which would determine which type of a ri a provided string is"""
+    # We assume that it is a URL and parse it. Depending on the result
+    # we might decide that it was something else ;)
```

Comment:
I'm fine with later ;-)
Am 09.06.2016 15:52 schrieb "Yaroslav Halchenko" <notifications@github.com>:

> In datalad/support/network.py
> <https://github.com/datalad/datalad/pull/500#discussion_r66444117>:
>
> > +
> > +
> > +# TODO:  make it consistent/clear at what stage % encoding/decoding happens!
> > +# now it is a mix!
> > +
> > +#
> > +# Useful functionality in requests.models
> > +#  utils.requote_uri -- quote/unquote cycle to guarantee consistent appearance
> > +#  RequestEncodingMixin._encode_params -- Will successfully encode parameters when passed as a dict or a list of ...
> > +#  PreparedRequest().prepare_url(url, params) -- nicely cares about url encodings etc
> > +#
> > +
> > +def _guess_ri_cls(ri):
> > +    """Factory function which would determine which type of a ri a provided string is"""
> > +    # We assume that it is a URL and parse it. Depending on the result
> > +    # we might decide that it was something else ;)
>
> and while "guessing" we might want to allow to specify the viable RIs to
> consider, so not all are tested. but again -- I think -- later
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/datalad/datalad/pull/500/files/3e36be3d8d3012fc07864658e65d37f776bdd784#r66444117>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKAw_fsaw7_p-0avubLaRORtigAmHTY3ks5qKBqygaJpZM4IuCQr>
> .
>